plugins/modules/asset.py

```diff
@@ -90,9 +103,58 @@
 )
 
 
+def validate_module_params(params):
+    if params["state"] == "present":
+        if not params['builds']:
+            return "builds must include at least one element"
+    return None
+
+
+def _sorted_builds(builds):
+    build_tuples = [(
+        b.get('sha512'),
+        b.get('url'),
+        b.get('headers', {}),
+        frozenset(b.get('filters', []) or []),
+    ) for b in builds]
+
+    return sorted(build_tuples, key=lambda x: (x[0]))
```

Comment:
Alternative implementation (using sets) that does not need sorting would be:

```
    return {(
        b.get('sha512'),
        b.get('url'),
        frozenset((b.get('headers', {}) or {}).items()),
        frozenset(b.get('filters', []) or []),
    ) for b in builds}
```

Now we can simply compare the results of this function using reqular `==` and `!=`.